### PR TITLE
closes #8917 introduce `—workdir` option for docker exec

### PR DESCRIPTION
**Pull request by ndeloof**
Signed-off-by: Nicolas De Loof <nicolas.deloof@gmail.com>

** Make sure all your commits include a signature generated with `git commit -s` **

**- What I did**
introduced `--workdir` option to docker exec command
NOTE : require https://github.com/moby/moby/pull/35661/

**- Description for the changelog**

Introduce --workdir option for docker exec to set process working directory

**- A picture of a cute animal (not mandatory but encouraged)**

8a2c3da5f14275a31e1376e6f7ad16cd--cute-horses-horse-love

### CODE REVIEW

File: cli/command/container/exec.go [hunk 1]
```diff
@@ -57,6 +58,7 @@ func NewExecCommand(dockerCli command.Cli) *cobra.Command {
 	flags.StringVarP(&options.user, "user", "u", "", "Username or UID (format: <name|uid>[:<group|gid>])")
 	flags.BoolVarP(&options.privileged, "privileged", "", false, "Give extended privileges to the command")
 	flags.VarP(&options.env, "env", "e", "Set environment variables")
+	flags.StringVarP(&options.workdir, "workdir", "w", "", "Working directory inside the container")
```
Comment (thaJeztah) [inline cli/command/container/exec.go]: Can you move this below the version annotation for `env`, and add a version annotation for the new option? The annotation is used to hide the flag when talking to an older daemon version (which doesn't yet support the feature)

```go
flags.SetAnnotation("env", "version", []string{"1.25"})
flags.StringVarP(&options.workdir, "workdir", "w", "", "Working directory inside the container")
flags.SetAnnotation("workdir", "version", []string{"1.35"})
```

File: vendor/github.com/docker/distribution/vendor.conf [hunk 2]
```diff
@@ -1,43 +0,0 @@
-github.com/Azure/azure-sdk-for-go 088007b3b08cc02b27f2eadfdcd870958460ce7e
```
Comment (thaJeztah) [inline vendor/github.com/docker/distribution/vendor.conf]: Wait, why did all the nested vendor.conf files disappear?

@ndeloof could it be that you used an older version of `vndr`?


I just ran `vndr` on the PR and got this change;


[revendor.patch.txt](https://github.com/docker/cli/files/1536213/revendor.patch.txt)

**Comment by ndeloof** [comment]
not sure about the cause for https://circleci.com/gh/docker/cli/7883 failure. 
`make vendor` is what I've used to update docker/docker to merge commit for engine-side WorkingDir support. So I would not expected it detects changes here

I ran `make -B vendor` on my working copy and didn't ha those reported changes.

**Comment by thaJeztah** [comment]
Oh, my comment crossed yours; I'm on Docker for Mac myself, and run `vndr` inside a container (`make -f docker.Makefile vendor`); perhaps you ran `vndr` outside of a container, and used the version that you had installed on your host.

This is the version that's used in our CI: https://github.com/docker/cli/blob/8abb2660434483edcb026d848ece34d74c3e75f1/dockerfiles/Dockerfile.dev#L6-L11

We pinned it to a specific commit, because there have been some updates in the past causing different results (like is seen here)

Let me know if you need help!

**Comment by ndeloof** [comment]
got it, thanks. @thaJeztah requested changes have been implemented AFAICT

**Pull request merged**